### PR TITLE
Add option to trigger crash during migration

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
As requested in #13854

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
@lukstbit I'm sorry but I don't understand your tag. There is no comment on question, how could I reply to anything.
@oakkitten, would you mind commenting, as I was trying to follow the idea you mentioned.

File: AnkiDroid/src/main/res/values/preferences.xml [hunk 1]
```diff
@@ -113,6 +113,7 @@
     <string name="card_browser_external_context_menu_key">card_browser_enable_external_context_menu</string>
     <string name="anki_card_external_context_menu_key">anki_card_enable_external_context_menu</string>
     <string name="pref_reset_languages_key">resetLanguages</string>
+    <string name="crash_during_essential_migration">crashDuringMainMigration</string>
```
Comment (oakkitten) [inline AnkiDroid/src/main/res/values/preferences.xml:116]: You've got “essential migration” and “main migration” in one line. I **strongly** suggest reusing terminology that we already have, throughout. Namely, “essential file migration”. So I'd go with “Simulate error during essential file migration”.

Change related names accordingly

**Comment by lukstbit** [comment]
I add that label because something is preventing the review, either tests are failing or there is a merge conflict(I think this had a merge conflict when I initially looked at it). That label becomes a notice for me(and I hope others) that it isn't yet ready for review so I don't repeatedly check it out.

**Comment by Arthur-Milchior** [comment]
@lukstbit May I advise checking the nature of the merge conflict?
You're right, I just had to rebase to resolve one. But, given the merge conflict was related to two imports appearing at the same place in the list of imports, I suspect that the code was ready for review. And even if you wanted to execute it to test locally, the resolution was a non-brainer

**Comment by Arthur-Milchior** [comment]
@oakkitten reworded/renamed

**Review by oakkitten — CHANGES_REQUESTED** [review]
I suggest using the phrase “Simulate error during essential file migration” (without the `s` and the period), as well as renaming *all* related keys and string values and constants and variables to mirror this phrase. Using a single phrase for everything greatly reduces confusion **and** it's so easy to do (and easy to search and everything), besides, you don't have to think, just rename everything related.

**Comment by Arthur-Milchior** [comment]
Done.

File: AnkiDroid/src/main/res/values/preferences.xml [hunk 2]
```diff
@@ -113,6 +113,7 @@
     <string name="card_browser_external_context_menu_key">card_browser_enable_external_context_menu</string>
     <string name="anki_card_external_context_menu_key">anki_card_enable_external_context_menu</string>
     <string name="pref_reset_languages_key">resetLanguages</string>
+    <string name="simulate_error_during_essential_file_migration">crashDuringEssentialMigration</string>
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/values/preferences.xml:116]:
```suggestion
    <string name="simulate_error_during_essential_file_migration_key">crashDuringEssentialMigration</string>
```

File: AnkiDroid/src/main/res/xml/preferences_dev_options.xml [hunk 3]
```diff
@@ -51,4 +51,12 @@
         android:title="@string/reset_onboarding"
         android:summary="@string/reset_onboarding_desc"
         android:key="@string/pref_reset_onboarding_key"/>
+    <PreferenceCategory
+        android:title="Migration crashes">
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/xml/preferences_dev_options.xml:55]:
```suggestion
        android:title="Crash migration">
```

File: AnkiDroid/src/main/res/xml/preferences_dev_options.xml [hunk 4]
```diff
@@ -51,4 +51,12 @@
...
+        <SwitchPreference
+            android:defaultValue="false"
+            android:title="Simulate error during essential file migration"
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/xml/preferences_dev_options.xml:58]:
```suggestion
            android:title="Simulate error during essential files migration"
```

**Review by BrayanDSO — APPROVED** [review]
I don't effectively care about the names and the keys. This is a dev option and we are post the deadline, so it only matters if it works. So I'm approving.

Reviewers' note: avoid merging this before #13969 as it will generate conflicts and #13969 is way over higher priority than this, so I'm even adding the "Blocked by dependency" label

File: AnkiDroid/src/main/res/xml/preferences_dev_options.xml (hunk 3 above)
Comment (oakkitten) [reply AnkiDroid/src/main/res/xml/preferences_dev_options.xml:55]: This doesn't crash the app

File: AnkiDroid/src/main/res/xml/preferences_dev_options.xml (hunk 4 above)
Comment (oakkitten) [reply AnkiDroid/src/main/res/xml/preferences_dev_options.xml:58]: Not sure about this one, it's “bird migration” not “birds migration”, but you can always say “migration of essential files”

**Pull request closed without merging**